openlibrary/plugins/importapi/import_validator.py

```diff
@@ -1,24 +1,185 @@
+        examples=['eng', 'fre', 'ger', 'tha'],
+    )
+
+
+class Isbn10(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^([0-9][- ]?){9}[0-9X]$')]
+
+
+class Isbn13(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^([0-9][- ]?){13}$')]
+
+
+class WorkKey(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^/works/OL[0-9]+W$')]
+
+
+class Language(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+    key: Annotated[str, StringConstraints(pattern=r'^/languages/[a-z]{3}$')] = Field(
+        ..., examples=['/languages/eng', '/languages/ger']
+    )
+
+
+class LcClassification(RootModel):
+    root: NonEmptyStr = Field(
+        ...,
+        description='The Library of Congress Classification number. See https://www.loc.gov/catdir/cpso/lcc.html We include the imprint date as the last four digits.',  # noqa: E501
+        examples=['BS571.5 .S68 1995', 'Z673.D62 C65 1994'],
+    )
+
+
+class Key(Enum):
+    field_type_link = '/type/link'
+
+
+class Type(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+    key: Key
+
+
+class Link(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+    url: AnyHttpUrl
+    title: NonEmptyStr
+    type: Optional[Type] = None
+
+
+class EntityType(Enum):
+    person = 'person'
+    org = 'org'
+    event = 'event'
 
 
 class Author(BaseModel):
-    name: NonEmptyStr
+    model_config = ConfigDict(extra="forbid")
+
+    name: NonEmptyStr = Field(examples=['Hubbard, Freeman H.', 'Joan Miró'])
+    personal_name: Optional[NonEmptyStr] = Field(
+        None,
+        description="Can be identical to 'name'. TODO: provide information on the intended difference.",
+    )
+    birth_date: Optional[NonEmptyStr] = Field(None, examples=[])
+    death_date: Optional[NonEmptyStr] = Field(None, examples=[])
+    entity_type: Optional[EntityType] = None
+    title: Optional[NonEmptyStr] = Field(None, examples=["duc d'Otrante"])
 
 
 class Book(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
     title: NonEmptyStr
+    subtitle: Optional[NonEmptyStr] = None
     source_records: NonEmptyList[NonEmptyStr]
-    authors: NonEmptyList[Author]
     publishers: NonEmptyList[NonEmptyStr]
-    publish_date: NonEmptyStr
+    authors: NonEmptyList[Author]
+    publish_date: NonEmptyStr = Field(
+        ..., examples=['1992', 'December 1992', '12 January 2002']
+    )
+    location: Optional[NonEmptyList[NonEmptyStr]] = None
+    publish_places: Optional[NonEmptyList[NonEmptyStr]] = None
+    number_of_pages: Optional[PositiveInt] = None
+    pagination: Optional[NonEmptyStr] = None
+    by_statement: Optional[NonEmptyStr] = None
+    description: Optional[NonEmptyStr] = None
+    publish_country: Optional[PublishCountry] = None
+    languages: Optional[list[LanguageCode]] = None
+    translated_from: Optional[list[LanguageCode]] = None
+    translation_of: Optional[NonEmptyStr] = None
```

Comment:
These `translated_from` and `translated_of` could do with examples.

I thought there was some discussion around chaninging `translation_of` to a work id to help with linking?

Reply:
Not sure.. I didn't notice..